connectors/src/connectors/confluence/index.ts

```diff
@@ -49,479 +49,516 @@ import { getAccessTokenFromNango } from "@connectors/lib/nango_helpers";
-  const connector = await ConnectorResource.fetchById(connectorId);
-  if (!connector) {
-    logger.error({ connectorId }, "Connector not found.");
-    return new Err(new Error("Connector not found"));
-  }
+  async stop(): Promise<Result<undefined, Error>> {
+    const res = await stopConfluenceSyncWorkflow(this.connectorId);
+    if (res.isErr()) {
+      return res;
+    }
 
-  await connector.markAsPaused();
-  const r = await stopConfluenceSyncWorkflow(connectorId);
-  if (r.isErr()) {
-    return r;
+    return new Ok(undefined);
   }
 
-  return new Ok(undefined);
-}
-
-export async function unpauseConfluenceConnector(
-  connectorId: ModelId
-): Promise<Result<undefined, Error>> {
-  const connector = await ConnectorResource.fetchById(connectorId);
-  if (!connector) {
-    logger.error({ connectorId }, "Connector not found.");
-    return new Err(new Error("Connector not found"));
-  }
+  async resume(): Promise<Result<undefined, Error>> {
+    try {
+      const connector = await ConnectorResource.fetchById(this.connectorId);
+      if (!connector) {
+        return new Err(
+          new Error(
+            `Confluence connector not found (connectorId: ${this.connectorId}`
+          )
+        );
+      }
 
-  await connector.markAsUnpaused();
-  const r = await launchConfluenceSyncWorkflow(connectorId, null);
-  if (r.isErr()) {
-    return r;
-  }
+      const connectorState = await ConfluenceConfiguration.findOne({
+        where: {
+          connectorId: connector.id,
+        },
+      });
+      if (!connectorState) {
+        return new Err(new Error("Confluence configuration not found"));
+      }
 
-  return new Ok(undefined);
-}
+      await launchConfluenceSyncWorkflow(connector.id, null);
 
-export async function cleanupConfluenceConnector(
-  connectorId: ModelId
-): Promise<Result<undefined, Error>> {
-  const connector = await ConnectorResource.fetchById(connectorId);
-  if (!connector) {
-    logger.error({ connectorId }, "Confluence connector not found.");
-    return new Err(new Error("Connector not found"));
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
   }
 
-  const nangoRes = await nangoDeleteConnection(
-    connector.connectionId,
-    getRequiredNangoConfluenceConnectorId()
-  );
-  if (nangoRes.isErr()) {
-    throw nangoRes.error;
+  async sync({
+    fromTs,
+  }: {
+    fromTs: number | null;
+  }): Promise<Result<string, Error>> {
+    return launchConfluenceSyncWorkflow(this.connectorId, fromTs);
   }
 
-  const res = await connector.delete();
-  if (res.isErr()) {
-    logger.error(
-      { connectorId, error: res.error },
-      "Error cleaning up Confluence connector."
-    );
-    return res;
-  }
+  async retrievePermissions({
+    parentInternalId,
+    filterPermission,
+  }: {
+    parentInternalId: string | null;
+    filterPermission: ConnectorPermission | null;
+    viewType: ContentNodesViewType;
```

Comment:
Does not seem used for Confluence, the type system is flexible enough that you can remove it if you don't use it, even for inherited methods.